### PR TITLE
Add some tests to app-permissions package

### PR DESCRIPTION
**Pull request by canadaduane**

### CODE REVIEW

File: packages/app-permissions/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  it('merges', async () => {
+    const app = readJsonFixture('grant_google_twitter.json')
+    const user = readJsonFixture('grant_google_mainframe.json')
+
+    const details = readJsonFixture('details_merged.json')
+
+    expect(mergeGrantsToDetails(app, user)).toEqual(details)
+  })
+
+  it('session grants are UNIONed', async () => {
```
Comment (PaulLeCam) [inline packages/app-permissions/__tests__/index.js]: No need for `async`

File: packages/app-permissions/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { readFileSync } from 'fs'
+import { EMPTY_DEFINITIONS, mergeGrantsToDetails } from '../lib'
+
+const readJsonFixture = filename => {
```
Comment (PaulLeCam) [inline packages/app-permissions/__tests__/index.js]: Why not simply include the test values in this file? It shouldn't need such a setup.

File: packages/app-permissions/src/index.js [hunk 3]
```diff
@@ -33,7 +33,12 @@ export const mergeGrantsToDetails = (
     ...app, // Manifest requirements overrides
     // Special case
     HTTPS_REQUEST: {
-      granted: app.HTTPS_REQUEST.granted.concat(user.HTTPS_REQUEST.granted),
+      granted: [
+        ...new Set([
```
Comment (PaulLeCam) [inline packages/app-permissions/src/index.js]: This shouldn't be done here, the `PermissionsGrants` themselves should be sanitised

File: packages/app-permissions/package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "build:js": "flow-remove-types src --out-dir lib",
     "build:flow": "flow-copy-source src lib",
     "build": "yarn clean && yarn build:js && yarn build:flow",
-    "test:types": "flow check"
+    "test:types": "flow check",
+    "test:unit": "yarn build && ../../node_modules/jest/bin/jest.js --setupFiles ../../jest.setup.js"
```
[the remark this answers is not in the document]
Comment (canadaduane) [reply packages/app-permissions/package.json]: How do you run tests for just one package?

File: packages/app-permissions/__tests__/index.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+describe('EMPTY_DEFINITIONS', () => {
+  it('has HTTPS_REQUEST', async () => {
+    expect(EMPTY_DEFINITIONS).toHaveProperty('HTTPS_REQUEST')
+  })
+})
+
+describe('mergeGrantsToDetails', () => {
+  it('merges', async () => {
```
[the remark this answers is not in the document]
Comment (canadaduane) [reply packages/app-permissions/__tests__/index.js]: Copy pasta! Thanks.

File: packages/app-permissions/__tests__/index.js (hunk 2 above)
Comment (canadaduane) [reply packages/app-permissions/__tests__/index.js]: I had a much larger test before, but pared it down. I think there may be other fixtures.

File: packages/app-permissions/src/index.js (hunk 3 above)
Comment (canadaduane) [reply packages/app-permissions/src/index.js]: I don't think I understand... whether or not the PermissionsGrants are sanitized, they still need a set union to remove duplicates when merging.

File: packages/app-permissions/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply packages/app-permissions/package.json]: I don't think you should, the point of having these modules in a monorepo is that there are cross-dependencies between them, so test shouldn't only run isolated but also make sure they are not breaking expectations from other modules.

File: packages/app-permissions/__tests__/index.js (hunk 2 above)
Comment (PaulLeCam) [reply packages/app-permissions/__tests__/index.js]: Please import the objects in the test cases for now, it makes things more readdable. Doing synchronous I/O in node is not ideal, even when it's just for tests.

File: packages/app-permissions/src/index.js (hunk 3 above)
Comment (PaulLeCam) [reply packages/app-permissions/src/index.js]: There should never be a domain present both in `app` and `user` `PermissionGrants`, if so it means there is an issue with the way they are managed.

File: packages/app-permissions/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (canadaduane) [reply packages/app-permissions/package.json]: My work-flow is usually to write tests and check just the test I'm writing, then when it's functioning as I expect, to loop back and run all tests. It's more efficient than always running all the tests.

File: packages/app-permissions/__tests__/index.js (hunk 2 above)
Comment (canadaduane) [reply packages/app-permissions/__tests__/index.js]: Ok

File: packages/app-permissions/src/index.js (hunk 3 above)
Comment (canadaduane) [reply packages/app-permissions/src/index.js]: I'll revert.

File: packages/app-permissions/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (mosic) [reply packages/app-permissions/package.json]: @canadaduane Just FYI, you can run individual tests or groups of tests like this:
```
jest -t "mergeGrantsToDetails"
```
or
```
jest -t "session grants"
```

[the remark this answers is not in the document]
Comment (canadaduane) [reply packages/app-permissions/package.json]: `yarn jest -t "mergeGrantsToDetails"` worked for me. Thanks @mosic !

**Comment by canadaduane** [comment]
Well, that was fun. Down to one test file.

**Comment by canadaduane** [comment]
Use PR#22 instead

**Pull request closed without merging**